### PR TITLE
fix infinite loop in src_overlay

### PR DESCRIPTION
**Pull request by lefessan**
This PR contains two commits:
* one to add debug stuff to help solving bugs by outputing in logs and reducing the size of these logs by detecting cycles
* one to fix the bug in src_overlay, that happens because the code is looking for an exact match, that actually does not exist

### CODE REVIEW

File: src/lsp/cobol_preproc/src_overlay.ml [hunk 1]
```diff
@@ -129,7 +140,13 @@ let join_limits: manager -> limit * limit -> srcloc = fun ctx (s, e) ->
       if e.pos_cnum  = e'.pos_cnum &&   (* compare only the fields that matter *)
          e.pos_fname = e'.pos_fname
       then (Links.replace ctx.cache s (loc, e); loc)
-      else try_cache_from ~loc @@ Links.find ctx.over_right_gap e'
+      else
+      if e.pos_cnum  = e'.pos_cnum &&   (* compare only the fields that matter *)
```
Comment (nberth) [inline src/lsp/cobol_preproc/src_overlay.ml:144]: Please don't merge this, this is not a proper fix (some `pos_cnum`s are not positive).

Comment (nberth) [reply src/lsp/cobol_preproc/src_overlay.ml]: (Not the same code I saw earlier)

Comment (lefessan) [reply src/lsp/cobol_preproc/src_overlay.ml]: fixed

File: src/lsp/cobol_preproc/src_overlay.ml [hunk 2]
```diff
@@ -129,7 +140,13 @@ let join_limits: manager -> limit * limit -> srcloc = fun ctx (s, e) ->
       if e.pos_cnum  = e'.pos_cnum &&   (* compare only the fields that matter *)
          e.pos_fname = e'.pos_fname
       then (Links.replace ctx.cache s (loc, e); loc)
-      else try_cache_from ~loc @@ Links.find ctx.over_right_gap e'
+      else
+      if e.pos_cnum  < e'.pos_cnum &&   (* detect that there is no exact match *)
```
Comment (nberth) [inline src/lsp/cobol_preproc/src_overlay.ml:144]: For a "cleaner" temporary fix (that would make the bug just "less frequent"), please guard this with `e.pos_cnum > 0`.

**Pull request merged**